src/Microsoft.ML.ImageAnalytics/MLImage.cs

```diff
@@ -0,0 +1,441 @@
+        private static readonly SKColorFilter _grayscaleColorMatrix = SKColorFilter.CreateColorMatrix(new float[]
+                                                                        {
+                                                                            0.3f, 0.59f, 0.11f, 0, 0,
+                                                                            0.3f, 0.59f, 0.11f, 0, 0,
+                                                                            0.3f, 0.59f, 0.11f, 0, 0,
+                                                                            0,    0,     0,     1, 0
+                                                                        });
+
+        internal MLImage CloneWithGrayscale()
+        {
+            ThrowInvalidOperationExceptionIfDisposed();
+
+            SKBitmap dst = new SKBitmap(_image.Width, _image.Height, isOpaque: true);
+            using SKPaint paint = new SKPaint()
+            {
+                ColorFilter = _grayscaleColorMatrix,
+                FilterQuality = SKFilterQuality.High
+            };
+
+            SKBitmap destBitmap = new SKBitmap(_image.Width, _image.Height, isOpaque: true);
+            using SKCanvas canvas = new SKCanvas(destBitmap);
+            canvas.DrawBitmap(_image, 0f, 0f, paint: paint);
+            return new MLImage(destBitmap);
+        }
+
+        private static readonly Dictionary<string, SKEncodedImageFormat> _extensionToEncodingFormat = new Dictionary<string, SKEncodedImageFormat>(StringComparer.OrdinalIgnoreCase)
+        {
+            { ".bmp", SKEncodedImageFormat.Bmp },
+            { ".png", SKEncodedImageFormat.Png },
+            { ".jpg", SKEncodedImageFormat.Jpeg },
+            { ".jpeg", SKEncodedImageFormat.Jpeg },
+            { ".gif", SKEncodedImageFormat.Gif },
+            { ".ico", SKEncodedImageFormat.Ico },
+            { ".astc", SKEncodedImageFormat.Astc },
+            { ".avif", SKEncodedImageFormat.Avif },
+            { ".dng", SKEncodedImageFormat.Dng },
+            { ".heif", SKEncodedImageFormat.Heif },
+            { ".ktx", SKEncodedImageFormat.Ktx },
+            { ".pkm", SKEncodedImageFormat.Pkm },
+            { ".wbmp", SKEncodedImageFormat.Wbmp },
+            { ".webp", SKEncodedImageFormat.Webp }
+        };
+
+        private void ThrowInvalidOperationExceptionIfDisposed()
+        {
+            if (_image is null)
+            {
+                throw new InvalidOperationException("Object is disposed.");
+            }
+        }
+    }
+
+    /// <summary>
+    /// The mode to decide how the image should be resized.
+    /// </summary>
+    public enum MLPixelFormat
+    {
+        /// <summary>
+        /// Pads the resized image to fit the bounds of its container.
+        /// </summary>
+        Unknown,
+
+        /// <summary>
+        /// Specifies that the format is 32 bits per pixel; 8 bits each are used for the blue, green, red, and alpha components.
```

Comment:
Might be helpful to mention byte order.

Reply:
I mentioned the color in by order but I'll make it explicit clear.

Reply:
Addressed.